pkg/solana/txm/pendingtx.go

```diff
@@ -84,48 +128,62 @@ func (c *pendingTxContext) Add(id uuid.UUID, sig solana.Signature) error {
 	if _, exists := c.sigToID[sig]; exists {
 		return errors.New("signature already exists")
 	}
```

Comment:
Think it's an optimization to do validations with just a read lock since state changes could be frequent depending on transaction throughput. But we have to recheck to be safe when switching to the write lock. I also wanted to maintain the flow from the existing in-memory storage methods.

Reply:
Ok, yeah that makes sense

Reply:
The optimization of handling a read lock and then a write lock is considerable? Or could we just use a write lock and make the things simpler

Reply:
This was an optimization that was built into the existing TXM's in-memory layer so I didn't want to cause any regressions if it was decided it was helpful before.